src/WebJobs.Script/Binding/WebJobsCoreScriptBindingProvider.cs

```diff
@@ -1,298 +0,0 @@
-            JToken value = null;
-            if (configSection != null)
-            {
-                if (configSection.TryGetValue("maxPollingInterval", out value))
-                {
-                    Config.Queues.MaxPollingInterval = TimeSpan.FromMilliseconds((int)value);
-                }
-                if (configSection.TryGetValue("batchSize", out value))
-                {
-                    Config.Queues.BatchSize = (int)value;
-                }
-                if (configSection.TryGetValue("maxDequeueCount", out value))
-                {
-                    Config.Queues.MaxDequeueCount = (int)value;
-                }
-                if (configSection.TryGetValue("newBatchThreshold", out value))
-                {
-                    Config.Queues.NewBatchThreshold = (int)value;
-                }
-            }
-
-            Config.UseScriptExtensions();
-        }
-
-        public override bool TryCreate(ScriptBindingContext context, out ScriptBinding binding)
-        {
-            binding = null;
-
-            if (string.Equals(context.Type, "table", StringComparison.OrdinalIgnoreCase))
-            {
-                binding = new TableScriptBinding(context);
-            }
-            else if (string.Compare(context.Type, "queueTrigger", StringComparison.OrdinalIgnoreCase) == 0 ||
-                string.Compare(context.Type, "queue", StringComparison.OrdinalIgnoreCase) == 0)
-            {
-                binding = new QueueScriptBinding(context);
-            }
-            else if (string.Compare(context.Type, "blobTrigger", StringComparison.OrdinalIgnoreCase) == 0 ||
-                     string.Compare(context.Type, "blob", StringComparison.OrdinalIgnoreCase) == 0)
-            {
-                binding = new BlobScriptBinding(context);
-            }
-            else if (string.Compare(context.Type, "httpTrigger", StringComparison.OrdinalIgnoreCase) == 0)
-            {
-                binding = new HttpScriptBinding(context);
-            }
-            else if (string.Compare(context.Type, "manualTrigger", StringComparison.OrdinalIgnoreCase) == 0)
-            {
-                binding = new ManualScriptBinding(context);
-            }
-
-            return binding != null;
-        }
-
-        private class HttpScriptBinding : ScriptBinding
-        {
-            public HttpScriptBinding(ScriptBindingContext context) : base(context)
-            {
-            }
-
-            public override Type DefaultType
-            {
-                get
-                {
-                    return typeof(HttpRequestMessage);
-                }
-            }
-
-            public override Collection<Attribute> GetAttributes()
-            {
-                Collection<Attribute> attributes = new Collection<Attribute>();
-
-                attributes.Add(new HttpTriggerAttribute
-                {
-                    RouteTemplate = Context.GetMetadataValue<string>("route")
```

Comment:
In the HttpTriggerAttribute, there's a new nested class that declares the mapping.  Again, that mapping only exists because we created an impedence mismatch (Route vs. RouteTemplate). New extensions won't have that, and so they'll work automatically without the extra mapping.